File: app/src/main/java/com/nipa/healthcareapp/services/MyFirebaseInstanceIDService.java
package com.nipa.healthcareapp.services;

import android.util.Log;
import androidx.annotation.NonNull; // For @NonNull on token parameter
import com.google.firebase.messaging.FirebaseMessagingService;

public class MyFirebaseInstanceIDService extends FirebaseMessagingService {

    private static final String TAG = "MyFirebaseInstanceIDService"; // Made static final

    @Override
    public void onNewToken(@NonNull String token) { // Added @NonNull
        super.onNewToken(token);
        Log.d(TAG, "New token: " + token);
        // Send token to your server
    }
}

File: app/src/main/java/com/nipa/healthcareapp/services/MyFirebaseInstanceIDService.kt
package com.nipa.healthcareapp.services

import android.util.Log
import com.google.firebase.messaging.FirebaseMessagingService
import com.nipa.healthcareapp.utils.Logger

/**
 * Handles Firebase Cloud Messaging token refresh events
 * Note: FirebaseInstanceIdService is deprecated, using FirebaseMessagingService instead
 */
class MyFirebaseInstanceIDService : FirebaseMessagingService() {

    private val TAG = "MyFirebaseInstanceIDService"

    override fun onNewToken(token: String) {
        super.onNewToken(token)
        // Called when the token is refreshed
        Log.d(TAG, "New token: $token")
        // Send token to your server
    }
}
